src/main/kotlin/org/rust/ide/actions/RustFmtFileAction.kt

```diff
@@ -0,0 +1,35 @@
+package org.rust.ide.actions
+
+import com.intellij.execution.ExecutionException
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.externalSystem.util.ExternalSystemUiUtil
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.openapi.roots.ModuleRootManager
+import com.intellij.openapi.ui.MessageType
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.util.PathUtil
+import org.rust.cargo.commands.Cargo
+import org.rust.cargo.project.pathToCargo
+import org.rust.cargo.project.util.getModules
+
+class RustFmtFileAction : DumbAwareAction() {
+    override fun actionPerformed(e: AnActionEvent?) {
+        val project = e?.project ?: return
+        val file = e?.getRequiredData(CommonDataKeys.VIRTUAL_FILE) ?: return
+
+        val module = project.getModules().firstOrNull() ?: return
+        val moduleManager = ModuleRootManager.getInstance(module)
+        val pathToCargo = moduleManager.sdk?.pathToCargo ?: return
+        val moduleDirectory = PathUtil.getParentPath(module.moduleFilePath)
+
+        val component = e?.getRequiredData(CommonDataKeys.EDITOR)?.component ?: return
+
+        try {
+            Cargo(pathToCargo, moduleDirectory).reformatFile(file.path)
+            VfsUtil.markDirtyAndRefresh(true, true, true, file);
```

Comment:
This line can be outside of the `try` block. It's better to minimize scope of `try`s